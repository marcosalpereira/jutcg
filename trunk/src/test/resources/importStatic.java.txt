package br.gov.esaf.sgc.qqpacote;

import static br.gov.esaf.scg.Sexo.MASCULINO;

public class Bean {
	private boolean masculino;

    public void foo(Sexo.MASCULINO sexo) {
        if(sexo == MASCULINO) {
        	masculino = true; 
        }
    }    
}