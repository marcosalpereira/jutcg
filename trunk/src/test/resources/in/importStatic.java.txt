package br.anypackage;

import static br.Genre.FEMALE;
import br.dao.Dao;

public class Business {
	@EJB
	private Dao dao;

    public List<Person> findWomen() {
    	return dao.find(FEMALE);
    }    
}