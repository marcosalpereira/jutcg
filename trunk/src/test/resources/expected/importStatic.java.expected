package br.anypackage;

import static br.Genre.FEMALE;
import br.dao.Dao;
import static org.easymock.EasyMock.*;
import org.junit.*;
import java.util.*;
import static org.junit.Assert.*;

/**
 * Testes unitarios para a classe {@link Business}.
 */
public class TestBusiness  {

/**
 * Classe sendo testada {@link Business}.
 */
private Business business;

/**
 * Configuracoes iniciais.
 */
@Before
public void setUp() {
  business = new Business();
}

/**
 * Cria o mock {@link Dao} e seta na classe sendo testada.
 * @return o mock criado
 */
private Dao criarMockDao() {
  Dao mock = createStrictMock(Dao.class);
  business.setDao(mock);
  return mock;
}

/**
 * Teste para o metodo {@link Business#findWomen()}.
 */
@Test
public void testFindWomen() {

// variaveis usadas
final List<Person> persons = Arrays.asList(new Person(1));

// Configurando mock
Dao dao = criarMockDao();
expect(dao.find(FEMALE))
  .andReturn(persons);
replay(dao);

// invocar metodo sendo testado
final List<Person> personsReal = business.findWomen();

// checar estados dos mocks
verify(dao);

// verificacoes do resultado do metodo sendo testado
assertEquals(persons, personsReal);
}
}
